build.gradle.kts

```diff
@@ -1,29 +1,150 @@
+        from("$rootDir/Firebase/FirebaseDatabase/FirebaseDatabase.framework")
+        into("$rootDir/firebase-database/src/iosMain/c_interop/modules/FirebaseDatabase.framework")
+    }
+
+    val copyFirebaseFirestore by creating(Copy::class){
+        dependsOn(copyFirebaseDatabase)
+        from("$rootDir/Firebase/FirebaseFirestore/FirebaseFirestore.framework")
+        into("$rootDir/firebase-database/src/iosMain/c_interop/modules/FirebaseFirestore.framework")
+    }
+
+    val copyFirebaseFunctions by creating(Copy::class){
+        dependsOn(copyFirebaseFirestore)
+        from("$rootDir/Firebase/FirebaseFunctions/FirebaseFunctions.framework")
+        into("$rootDir/firebase-database/src/iosMain/c_interop/modules/FirebaseFunctions.framework")
+    }
+
+    val copyAllFirebaseFrameworks by creating(Copy::class){
+        dependsOn(copyFirebaseFunctions)
+        from("$rootDir/Firebase/FirebaseAnalytics/FirebaseCore.framework")
+        into("$rootDir/firebase-app/src/iosMain/c_interop/modules/FirebaseCore.framework")
+    }
+
+}
+
+
+
 subprojects {
 
-    group = "dev.teamhub.firebase"
+    apply(plugin="maven-publish")
+
+    group = "dev.gitlive"
+    version = "0.1.0-beta"
 
     repositories {
         mavenLocal()
         mavenCentral()
         google()
         jcenter()
+        maven {
+            name = "github"
+            url = uri("https://maven.pkg.github.com/gitliveapp/firebase-java")
+            credentials {
+                username = project.property("gpr.user") as String
+                password = project.property("gpr.key") as String
+            }
+        }
+    }
+
+
+    tasks {
+
+        val copyReadMe by registering(Copy::class) {
+            from(rootProject.file("README.md"))
+            into(file("$buildDir/node_module"))
+        }
+
+        val copyPackageJson by registering(Copy::class) {
+            from(file("package.json"))
+            into(file("$buildDir/node_module"))
+        }
+
+        val copyJS by registering {
+            doLast {
+                val from = File("$buildDir/classes/kotlin/js/main/${project.name}.js")
+                val into = File("$buildDir/node_module/${project.name}.js")
+                into.createNewFile()
+                into.writeText(from.readText()
+                    .replace("require('firebase-", "require('@gitlive/firebase-")
+//                .replace("require('kotlinx-serialization-kotlinx-serialization-runtime')", "require('@gitlive/kotlinx-serialization-runtime')")
+                )
+            }
+        }
+
+        val copySourceMap by registering(Copy::class) {
+            from(file("$buildDir/classes/kotlin/js/main/${project.name}.js.map"))
+            into(file("$buildDir/node_module"))
+        }
+
+
+        val publishToNpm by creating(Exec::class) {
+            
+            if(!File(rootDir, "Firebase").exists()) {
+                dependsOn(
+                    rootProject.tasks.named("copyAllFirebaseFrameworks").get(),
```

Comment:
seperate these out as they are related one is ios while other is js, and the task is publishToNpm